webpack/assets/javascripts/react_app/redux/actions/notifications/index.js

```diff
@@ -19,25 +19,25 @@ const getNotifications = url => (dispatch) => {
 
   if (isDocumentVisible) {
     API.get(url)
-      .done(onGetNotificationsSuccess)
-      .fail(onGetNotificationsFailed)
-      .always(triggerPolling);
+      .then(onGetNotificationsSuccess)
+      .catch(onGetNotificationsFailed)
```

Comment:
same thing here based on comment above... lot of promises...

Reply:
What is wrong with promises in API calls?

Reply:
I think the syntax here is much cleaner, personally, but will leave it to you to decide it ;)

https://redux-saga.js.org/docs/advanced/Testing.html